### DISCUSSION TITLE
Cloth simulation using PhysX or NvCloth

### DISCUSSION THREAD

**Discussion by Hochheilige01** [NVIDIA-Omniverse/PhysX#328] [General]
Hello there! 
I wonder is it possible to simulate cloth same as it works in old PhysX 2. PhysX 2 had NxCloth that supports cloth tearing, attaching to other physics shapes, collisions and raycasts. But I can't see any of this functionality in NvCloth or in modern PhysX.

As far as I understand in PhysX 2 we just can pass some NxMeshData, creating NxCloth, that will store information about vertices and indices of our cloth mesh. And along with that there are cloth flags for example NX_CLF_TEARABLE (the thing I'm looking for in NvCloth).  
So in NvCloth or PhysX 5 (I saw a SnippetPBDCloth sample) we have some kind of a particle system that creates based on mesh points. But how to deal with tearing using it? Is there something I missed in documentation?

And I also interested in making raycast from cloth as it was available in PhysX 2 and attach cloth to PhysX shapes?

Oh, and one more question. Is there a way to debug NvCloth? I mean with PhysX I use PVD, should it work with NvCloth as well?

**Comment by sschirm** [comment]
Hi @Hochheilige01,
 
Unfortunately, the cloth feature in PhysX 5 is not as feature complete as NvCloth was, indeed. Further the particle cloth feature you have been looking at is deprecated. We introduced a new PxDeformableSurface feature which will be released soon. However, it's won't support tearing and scene queries (raycast) neither. We are focusing on behavior, and might get to some of these features later, if there is significant demand.

Cheers, 
Simon
